update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,351 @@
+        var publication = insertPublicationWithoutDoiRequest(
+            getFixedClockWithDefaultTimeZone(mockOneHourBefore));
+
+        ApiUpdateDoiRequest updateDoiRequest = generateValidApiUpdateDoiRequest(publication.getIdentifier().toString());
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, publication.getOwner());
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+
+        assertThat(details.getDetail(),
+            containsString("You must initiate creation of a DoiRequest before you can update it."));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestMissingDoiRequest() throws IOException {
+        var publication = insertPublicationWithoutDoiRequest(
+            getFixedClockWithDefaultTimeZone(mockOneHourBefore));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, publication.getOwner());
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+
+        assertThat(details.getDetail(),
+            containsString("You must request changes to do"));
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenInputUsernameIsNotThePublicationOwner()
+        throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(DynamoDBDoiRequestsService.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            getFixedClockWithDefaultTimeZone(mockOneHourBefore));
+
+        ApiUpdateDoiRequest updateDoiRequest = generateValidApiUpdateDoiRequest(publication.getIdentifier().toString());
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, INVALID_USERNAME);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_FORBIDDEN)));
+
+        assertThatProblemDetailsDoesNotRevealSensitiveInformation(details, INVALID_USERNAME,
+            validUsername(publication));
+
+        assertThatLogsContainReasonForForbiddenMessage(appender, publication);
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenUsernameIsNotInRequestContext()
+        throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(UpdateDoiRequestHandler.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            getFixedClockWithDefaultTimeZone(mockOneHourBefore));
+
+        ApiUpdateDoiRequest updateDoiRequest = generateValidApiUpdateDoiRequest(publication.getIdentifier().toString());
+
```

Comment:
Hide the preceding details in a descriptively named method

Reply:
Explain to me on slack. :-)